### PR TITLE
Fixes issues #201, #202

### PR DESCRIPTION
**Pull request by rjones0**
Please review. I appreciate use of a constantly running timer and dll interop may be surprising. However, these were the best mechanisms I could find and I think the behaviour is certainly improved by handling this issue, so potentially justified. I could look at only running the timer when Map Point tool is enabled, if this is thought worthwhile.

I did some brief testing of performance and saw a possible moderate increase in CPU usage when this fix is in place, by circling the cursor rapidly within the map window, but nothing worrying (perhaps peaking at 15% CPU usage, not definitely attributable to the timer tick code without further testing).

### CODE REVIEW

File: source/DistanceAndDirection/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs [hunk 1]
```diff
@@ -72,6 +100,22 @@ public TabBaseViewModel()
                 RaisePropertyChanged(() => Point2Formatted);
             });
 
+            // Kick off monitoring of mouse position via a timer
+            System.Timers.Timer mouseCheckTimer = new System.Timers.Timer();
```
Comment (kgonzago) [inline source/DistanceAndDirection/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs]: @rjones0 Your logic for getting mouse position is solid but I'm uncertain if using a timer is best in this situation. Please look at accepted anwsers in these links for possible workarounds: http://gis.stackexchange.com/questions/164739/handling-arcmap-closing-event-in-addin-to-save-edits-automatically and http://stackoverflow.com/questions/6036918/how-to-forward-messages-eg-mouse-wheel-to-another-control-without-stealing-fo.

**Comment by kgonzago** [comment]
@rjones0 Were you able to look at the links I supplied in my review? I'm still iffy on using a timer to check mouse movements. What are your thoughts?

**Comment by rjones0** [comment]
@kgonzago I did have a good look at this a few days ago and didn't find the answer that time, so was leaving my subconcious to chew on it for a few days whilst working on other issues before going back. No light bulb yet though.

I couldn't see that I could get what I needed from those links i.e. notification of the mouse leaving the map control - for the first link I believe the hwnd returned by the map control was not compatible with usage in the link, and for the second it seemed like I needed a WinForms control somewhere in the mix and I wasn't sure how to make use of this with the available ArcMap/WPF components. Please let me know if I am missing something. I would usually have updated the case with results of my investigations but was remiss this time for some reason, apologies.

Agreed a timer is not ideal.

One idea that occured to me was to use the event fired by mouse over the addin form to clear the text box, which would achieve almost the same as with the timer, with the difference that the text box would not be cleared strictly when the mouse left the map control. End result would be the same though.

**Comment by rjones0** [comment]
Confirmed that the method outlined in first link (http://gis.stackexchange.com/questions/164739/handling-arcmap-closing-event-in-addin-to-save-edits-automatically) works, i.e. subclass System.Windows.Forms.NativeWindow and override WndProc method, grabbing the message passed to WndProc and responding based on type of message.

However, this solution is not perfect - there are a variety of messages but none of them reliably indicate a mouse out event for the map control. Therefore it's necessary to catch several different message types, but the problem with this approach is that in using a wider net we catch too much, leading to increased CPU cycles as CheckMousePosition is run too often. I will have a look to see if there is more detail in the messages that I could utilise.

**Comment by rjones0** [comment]
Attempted to catch mouse position as described here, but ArcMap crashes despite attempting to place code in each of MapPoint and TabBaseViewModel:

https://social.msdn.microsoft.com/Forums/vstudio/en-US/e66cd9b4-93a9-4c6d-b61a-06013981e3b0/capture-mouse-events-no-matter-what-window-is-in-focus-more?forum=csharpgeneral

**Comment by rjones0** [comment]
Behavior is as designed.

**Pull request closed without merging**